### DISCUSSION TITLE
Feature in fragment does not appear in parent project

### DISCUSSION THREAD

**Discussion by mikehenselersimfront** [wixtoolset/issues#7794] [Questions]
create fragment-project that includes components/files and feature 
compile fragment-project to wixlib
add fragment-project.wixlib into parent.msi compile

run new installer, or open msi with orca
observe: feature from fragment project does not appear. no files from fragment project have been added to installer.

I would expect just adding the fragment-project.wixlib to parent compile would include the fragment features automatically.

workaround:
create feature in parent project and add component refs for components in fragment-project 
or add FeatureRef to feature in fragment project

**Comment by barnson** [comment]
WiX only links in resources that have a reference from the parent product.

**Comment by mikehenselersimfront** [comment]
yes, this has been my experience, 
i was hoping it would not be required to add references and just adding the fragment wixlib would be sufficient.
thnks

**Reply by robmen** [reply]
If the WiX Toolset worked as you hoped, you'd always get all the code from any referenced .wixlib. That design would require us to ship a few hundred tiny .wixlibs (not an exaggeration) so you would not experience massive bloat of unused behavior in your MSI anytime you used one of the WiX extensions.

I don't think you'd _actually_ like the world you hoped for. 😄

**Comment by mikehenselersimfront** [comment]
Thanks Rob for explaining it.
